csharp/ql/src/semmle/code/csharp/dataflow/internal/rangeanalysis/RangeUtils.qll

```diff
@@ -67,40 +67,26 @@ predicate valueFlowStep(Expr e2, Expr e1, int delta) {
   or
   e2.(PreDecrExpr).getOperand() = e1 and delta = -1
   or
-  // exists(ArrayCreationExpr a |
-  //   arrayLengthDef(e2, a) and
-  //   a.getDimension(0) = e1 and
-  //   delta = 0
-  // )
-  // or
-  exists(Expr x |
+  exists(ConstantIntegerExpr x |
     e2.(AddExpr).getAnOperand() = e1 and
     e2.(AddExpr).getAnOperand() = x and
     not e1 = x
     or
-    exists(AssignAddExpr add | add = e2 |
-      add.getLValue() = e1 and add.getRValue() = x
-      or
-      add.getLValue() = x and add.getRValue() = e1
-    )
+    valueFlowStep(e2.(AssignOperation).getExpandedAssignment(), e1, delta)
```

Comment:
Sorry, what I meant was to put this disjunct up to the top-level of `valueFlowStep`.